### PR TITLE
Feature/user repo refactor

### PR DESCRIPTION
**Pull request by itsnameissteven**
### Description

* Is this a feature or a fix?
This is a fix
* Where should the reviewer start?
Review the User-test then head over to user-repository test. Finally review the UserRepo class
* What functionalities do these changes effect? Why were these changes made? What was the motivation behind these changes?
No functionality changed
* What outside features or research did you use to implement this fix?
n/a
* How should this be tested?
Run npm test. Also check out the app to make sure everything displays as expected.

### Relevant Screen Shots
***
#### Please review considerations below:
- [X] Follows Turing Style Guidelines
- [X] Changes generate no new warnings or errors
- [X] Checked code for and corrected any spelling errors
- [ ] README has been updated(if applicable)

### CODE REVIEW

File: src/Sleep.js [hunk 1]
```diff
@@ -39,26 +39,26 @@ class Sleep {
   }
   determineBestSleepers(date, userRepo) {
     let timeline = userRepo.chooseWeekDataForAllUsers(this.sleepData, date);
-    let userSleepObject = userRepo.isolateUsernameAndRelevantData(this.sleepData, date, 'sleepQuality', timeline);
+    let userSleepObject = userRepo.isolateUsernameAndRelevantData('sleepQuality', timeline);
```
Comment (aemiers) [inline src/Sleep.js:42]: Nice job refactoring all these params

File: test/user-test.js [hunk 2]
```diff
@@ -114,10 +40,35 @@ describe('User', function() {
       dailyStepGoal: 7000,
       friends: [1, 2, 3]
     });
+
```
Comment (alexthompson207) [inline test/user-test.js:43]: Maybe change these to ES6 Arrow Functions?

File: src/User-repo.js [hunk 3]
```diff
@@ -2,44 +2,56 @@ class UserRepo {
...
     return this.users.find((user) => id === user.id);
   };
+
   getDataFromUserID(id, dataSet) {
     return dataSet.filter((userData) => id === userData.userID);
   };
+
   calculateAverageStepGoal() {
-    var totalStepGoal = this.users.reduce((sumSoFar, data) => {
+    const totalStepGoal = this.users.reduce((sumSoFar, data) => {
```
Comment (aemiers) [inline src/User-repo.js:15]: Nice const usage!

File: test/userrepo-test.js [hunk 4]
```diff
@@ -1,16 +1,20 @@
 import { expect } from 'chai';
-
 import UserRepo from '../src/User-repo';
 import User from '../src/User';
 
-
-describe('User Repo', function() {
+describe('User Repo', () => {
   let user1;
```
Comment (alexthompson207) [inline test/userrepo-test.js]: I think you can write this on one line: let user1, user2, user3, ....... user4000;

File: test/user-test.js [hunk 5]
```diff
@@ -114,10 +40,35 @@ describe('User', function() {
...
+    expect(user2.id).to.equal(2);
+    expect(user2.name).to.equal("Allie McCarthy");
+  });
+
+  it('should return users first name', function() {
+    expect(user2.getFirstName()).to.equal("Allie");
+  });
+
+  it('should return list of friend names from user repository', function() {
     const users = [user1, user2, user3, user4];
```
Comment (aemiers) [inline test/user-test.js:72]: Do we really need 4 users here?

File: test/user-test.js [hunk 6]
```diff
@@ -4,78 +4,6 @@ import UserRepo from '../src/User-repo';
 import User from '../src/User';
 
 describe('User', function() {
-
```
Comment (alexthompson207) [inline test/user-test.js:7]: Maybe us a forEach to clean this up?

**Review by alexthompson207 — COMMENTED** [review]
Awesome job refactoring these methods and test - way to go! Also great use of `const`. I add a few comments on areas that could be refactored.

File: test/user-test.js [hunk 7]
```diff
@@ -114,10 +40,35 @@ describe('User', function() {
...
       friends: [1, 2, 3]
     });
+
+
+  it('should be a function', () => {
+    expect(User).to.be.a('function');
+    expect(user1).to.be.an.instanceof(User);
+  });
+
+  it('should take a user data object', () => {
```
Comment (alexthompson207) [inline test/user-test.js:50]: Maybe turn these expects into 1 line, i.e expect(user1).to.deep.equal({object})

**Comment by alexthompson207** [comment]
Nice work Steven, we appreciate you!

**Pull request merged**